bin/utils/utils.py

```diff
@@ -0,0 +1,350 @@
+import subprocess
+import logging
+import re
+import os
+import socket
+import tempfile
+from distutils import dir_util
+
+PROJECT_NAME_PREFIX = "d2docker"
```

Comment:
Really minor, but ``d2-docker`` makes more sense to me as prefix or just ``d2``.

Reply:
Yep, that's what I had. All was fine until I tested in docker-compose v1.17, which strips all non-alphanumeric chars from the name :( 1.24 does not, because people complained. At the moment, the container name must be predictable so we can use it in `docker` commands, but I'll check if we can use the labels to extract those names instead of using them directly.